### ISSUE TITLE
More informative error message in the Ipopt interface when the Jacobian or Hessian get NaN or inf entries

### ISSUE THREAD

**Issue by jaeandersson** [casadi/casadi#684]
The Ipopt interface now only reports:

```
IpoptInternal::jac_g: NaN or Inf detected.

Error evaluating Jacobian of equality constraints at user provided starting point.
```

It would be more informative for the user if the index of the (first encountered NaN/inf) entry is also reported. Same goes for other calculated quantities.

**Comment by petlist** [comment]
Hello Joel,

Is implementing this functionality planned in the nearest future? Is there a workaround to get the point where NaN or Inf were detected in v3.0? In my case IPOPT fails after 940 iterations with this message.

**Comment by jaeandersson** [comment]
I believe that it has already been implemented, in version 3.1. So try upgrading to that version - syntax differences to 3.0 are minimal.
